TLM/TLM/Manager/Impl/ExtSegmentManager.cs

```diff
@@ -67,10 +67,29 @@ public class ExtSegmentManager
             }
         }
 
+        public bool IsLaneAndItsSegmentValid(uint laneId) {
+            return IsLaneValid(laneId)
+                && IsValid(Singleton<NetManager>.instance.m_lanes.m_buffer[laneId].m_segment);
+        }
+
         public bool IsValid(ushort segmentId) {
             return Constants.ServiceFactory.NetService.IsSegmentValid(segmentId);
         }
 
+        /// <summary>
+        /// Check if a lane id is valid.
+        /// </summary>
+        ///
+        /// <param name="laneId">The id of the lane to check.</param>
+        ///
+        /// <returns>Returns <c>true</c> if valid, otherwise <c>false</c>.</returns>
+        public bool IsLaneValid(uint laneId) {
+            var createdDeleted = Singleton<NetManager>.instance.m_lanes.m_buffer[laneId].m_flags
```

Comment:
[comment]
you can use `laneId.ToLane()` extension method. 
its theoretically faster because it caches the array (but in practice I don't think it makes a difference).

Reply:
[low priority caveat]
this method should be static.
Also it makes sense to make it an extension method for NetLane
`bool IsLaneValid(this ref NetLane lane)`

Reply:
Yeah, I did the most simple refactoring first with a low probability of introducing bugs in the process.
The previous code called Singleton<NetManager>.instance.m_lanes.m_buffer[laneId], so I call it too.
Next steps would be to improve on that.
Instead of using ToLane() I'd rather make the method take a ref NetLane instead. 
a) it would make it pure with all the benefits we get from that.
b) we would save a lot of calls to ToLane() / ToSegment() / ToWhatever() because often the caller has access to the NetLane / NetSegment / Whatever already anyway.